ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13131,7 +13131,8 @@ public void validate() throws SemanticException {
           LOG.debug("validated " + usedp.getName());
           LOG.debug(usedp.getTable().getTableName());
           WriteEntity.WriteType writeType = writeEntity.getWriteType();
-          if (writeType != WriteType.UPDATE && writeType != WriteType.DELETE) {
+          if (writeType != WriteType.UPDATE && writeType != WriteType.DELETE && writeType != WriteType.DDL_NO_LOCK
```

Comment:
Why we think that the WriteType defines whether we need to check for archived parts, or not?

Reply:
I am fine with removing this completely for "transactionalInQuery", given this is never used.

Reply:
Yeah, checking `AcidUtils.isTransactionalTable(tbl)` seems more reasonable than depending on the `WriteType`